lib/compress.js

```diff
@@ -1435,6 +1436,13 @@ merge(Compressor.prototype, {
                         });
                         return true;
                     }
+                    if (node instanceof AST_Assign && node.operator == "="
+                        && node.left instanceof AST_SymbolRef
+                        && scope === self
+                        && assign_as_unused) {
```

Comment:
Would you mind making `assign_as_unused` the first condition in the `if`? It's faster if `assign_as_unused` is not in effect, and the same speed otherwise.